hazelcast/src/main/java/com/hazelcast/query/PagingPredicate.java

```diff
@@ -71,310 +56,64 @@
-     */
-    void setAnchor(int page, Map.Entry anchor) {
-        SimpleImmutableEntry anchorEntry = new SimpleImmutableEntry(page, anchor);
-        int anchorCount = anchorList.size();
-        if (page < anchorCount) {
-            anchorList.set(page, anchorEntry);
-        } else if (page == anchorCount) {
-            anchorList.add(anchorEntry);
-        } else {
-            throw new IllegalArgumentException("Anchor index is not correct, expected: " + page + " found: " + anchorCount);
-        }
-    }
-
-    /**
-     * After each query, an anchor entry is set for that page. see {@link #setAnchor(int, Map.Entry)}}
-     * For the next query user may set an arbitrary page. see {@link #setPage(int)}
-     * for example: user queried first 5 pages which means first 5 anchor is available
-     * if the next query is for the 10th page then the nearest anchor belongs to page 5
-     * but if the next query is for the 3nd page then the nearest anchor belongs to page 2
-     *
-     * @return nearest anchored entry for current page
-     */
-    Map.Entry<Integer, Map.Entry> getNearestAnchorEntry() {
-        int anchorCount = anchorList.size();
-        if (page == 0 || anchorCount == 0) {
-            return NULL_ANCHOR;
-        }
-
-        Map.Entry anchoredEntry;
-        if (page < anchorCount) {
-            anchoredEntry = anchorList.get(page - 1);
-        } else {
-            anchoredEntry = anchorList.get(anchorCount - 1);
-        }
-        return anchoredEntry;
-    }
-
-
-    @Override
-    public void writeData(ObjectDataOutput out) throws IOException {
-        out.writeObject(predicate);
-        out.writeObject(comparator);
-        out.writeInt(page);
-        out.writeInt(pageSize);
-        out.writeUTF(iterationType.name());
-        out.writeInt(anchorList.size());
-        for (Map.Entry<Integer, Map.Entry<K, V>> anchor : anchorList) {
-            out.writeInt(anchor.getKey());
-            Map.Entry<K, V> anchorEntry = anchor.getValue();
-            out.writeObject(anchorEntry.getKey());
-            out.writeObject(anchorEntry.getValue());
-        }
-    }
+    Map.Entry<K, V> getAnchor();
 
-    @Override
-    public void readData(ObjectDataInput in) throws IOException {
-        predicate = in.readObject();
-        comparator = in.readObject();
-        page = in.readInt();
-        pageSize = in.readInt();
-        iterationType = IterationType.valueOf(in.readUTF());
-        int size = in.readInt();
-        anchorList = new ArrayList<Map.Entry<Integer, Map.Entry<K, V>>>(size);
-        for (int i = 0; i < size; i++) {
-            int anchorPage = in.readInt();
-            Object anchorKey = in.readObject();
-            Object anchorValue = in.readObject();
-            Map.Entry anchorEntry = new SimpleImmutableEntry(anchorKey, anchorValue);
-            anchorList.add(new SimpleImmutableEntry<Integer, Map.Entry<K, V>>(anchorPage, anchorEntry));
-        }
-    }
+    IterationType getIterationType();
```

Comment:
These two (`getIterationType` and `setIterationType`) are kinda internal and provide nothing useful to a user. I think it's better to remove them from the interface and provide access from `PagingPredicateAccessor` only.

BTW, `getAnchor` also looks like a good candidate for removal, but I'm not sure about it. @jerrinot WDYT?

Reply:
I was wondering about the iteraton type, too. I will take a look at it.